src/components/DropzoneAreaFunc.js

```diff
@@ -0,0 +1,486 @@
+
+          return {
+            file,
+            data,
+          };
+        })
+      );
+
+      setFileObjects([].concat(fileObjects, fileObjs));
+      if (onChange) {
+        onChange(fileObjects.map((fileObject) => fileObject.file));
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const handleSnackbarState = (shouldOpen, message, variant) => {
+    setOpenSnackBar(shouldOpen);
+    setSnackbarMessage(message);
+    setSnackbarVariant(variant);
+  };
+
+  const handleDropAccepted = async (acceptedFiles, evt) => {
+    if (
+      filesLimit > 1 &&
+      fileObjects.length + acceptedFiles.length > filesLimit
+    ) {
+      handleSnackbarState(true, getFileLimitExceedMessage(filesLimit), "error");
+      return;
+    }
+
+    // Notify Drop event
+    if (onDrop) {
+      onDrop(acceptedFiles, evt);
+    }
+
+    // Retrieve fileObjects data
+    const fileObjs = await Promise.all(
+      acceptedFiles.map(async (file) => {
+        const data = await readFile(file);
+        return {
+          file,
+          data,
+        };
+      })
+    );
+
+    // Display message
+    const message = fileObjs.reduce(
+      (msg, fileObj) => msg + getFileAddedMessage(fileObj.file.name),
+      ""
+    );
+
+    handleSnackbarState(true, message, "success");
+
+    // Update component state
+    const concatedFiles = [...fileObjects, ...fileObjs];
+
+    setFileObjects(concatedFiles);
+
+    if (onChange) {
+      onChange(concatedFiles.map((fileObject) => fileObject.file));
+    }
+  };
+
+  const handleDropRejected = (rejectedFiles, evt) => {
+    let message = "";
+    rejectedFiles.forEach((rejectedFile) => {
+      message = getDropRejectMessage(rejectedFile, acceptedFiles, maxFileSize);
+    });
+
+    if (onDropRejected) {
+      onDropRejected(rejectedFiles, evt);
+    }
+
+    handleSnackbarState(true, message, "error");
+  };
+
+  const handleRemove = (fileIndex) => (event) => {
+    event.stopPropagation();
+
+    // Find removed fileObject
+    const removedFileObj = fileObjects.filter((fileObject, i) => {
+      return i === fileIndex;
+    })[0];
+    // Calculate remaining fileObjects array
+    const remainingFileObjs = fileObjects.filter((fileObject, i) => {
+      return i !== fileIndex;
+    });
+
+    setFileObjects(remainingFileObjs);
+
+    if (onDelete) {
+      onDelete(removedFileObj.file);
+    }
+
+    if (onChange) {
+      onChange(remainingFileObjs.map((fileObject) => fileObject.file));
+    }
+
+    handleSnackbarState(
+      true,
+      getFileRemovedMessage(removedFileObj.file.name),
+      "info"
+    );
+  };
+
+  const handleCloseSnackbar = () => {
+    setOpenSnackBar(false);
+  };
+
+  const acceptFiles = acceptedFiles?.join(",");
+  const isMultiple = filesLimit > 1;
+  const previewsVisible = showPreviews && fileObjects.length > 0;
+  const previewsInDropzoneVisible =
+    showPreviewsInDropzone && fileObjects.length > 0;
+
+  return (
+    <Fragment>
```

Comment:
Since we are using hooks we are now on the version of react where we can use the short hand syntax for fragments which is the <> </> notation, should we go for this while we're at it?

Reply:
I agree with you, we can (and should) use the latest syntax specifications supported by `React 16.8+`